### DISCUSSION TITLE
How to wrap a Client only Component?

### DISCUSSION THREAD

**Discussion by aheissenberger** [wakujs/waku#453] [Q&A] [answered]
**Goal:** load the component only in the browser

In other frameworks this is possible by lazy loading the component, but here the lazy loaded component will still be requested from the RSC API. Currently this breaks my code as the RSC tries to run the code on the server and fails with browser dependencies as `document` accessed by some used frontend libraries.

I think loading lazy loaded components with RSC is great but please provide an escape hatch or wrapper to load some components only in the browser.

**Comment by dai-shi** [comment, marked as answer]
Hi, are you talking about SSR? (= is it only the case with `--with-ssr`?)

This is often an escape hatch.
https://github.com/dai-shi/waku/blob/458e69da20b791d4433f1be6361d40564f64c1b9/packages/website/src/components/client-only.tsx#L10-L22
